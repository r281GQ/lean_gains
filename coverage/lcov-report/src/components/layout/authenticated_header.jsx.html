<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/layout/authenticated_header.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/layout</a> authenticated_header.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/3</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { Toolbar } from 'material-ui';
&nbsp;
import RouterNavigation from './router_navigation';
import Main from './main';
import Exercises from './exercises';
import User from './user';
import Macros from './macros';
import SaveIndicator from './save_indicator';
&nbsp;
&nbsp;
&nbsp;
const AuthenticatedHeader = <span class="fstat-no" title="function not covered" >({</span>
  openSideBar,
  exercises,
  isFetching,
  isLoading,
  userName,
  photo,
  todaysMacros,
  goBack,
  goForward
}) =&gt;
<span class="cstat-no" title="statement not covered" >  &lt;Toolbar style={{ backgroundColor: '#757575', color: '#EEEEEE' }}&gt;</span>
    &lt;Main openSideBar={openSideBar} /&gt;
    &lt;RouterNavigation goBack={goBack} goForward={goForward} /&gt;
    &lt;Exercises exercises={exercises} /&gt;
    &lt;Macros todaysMacros={todaysMacros} isFetching={isFetching} /&gt;
    &lt;SaveIndicator isLoading={isLoading} /&gt;
    &lt;User userName={userName} photo={photo} /&gt;
  &lt;/Toolbar&gt;;
&nbsp;
AuthenticatedHeader.propTypes = {
  openSideBar: PropTypes.func,
  exercises: PropTypes.arrayOf(PropTypes.string),
  isFetching: PropTypes.bool,
  isLoading: PropTypes.bool,
  userName: PropTypes.string,
  photo: PropTypes.string,
  todaysMacros: PropTypes.object
};
&nbsp;
export default AuthenticatedHeader;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
