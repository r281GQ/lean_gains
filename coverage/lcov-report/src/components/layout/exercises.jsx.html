<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/layout/exercises.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/layout</a> exercises.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/4</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import * as _ from 'lodash';
import {
  IconButton,
  ToolbarGroup,
  ToolbarTitle,
  MenuItem,
  IconMenu
} from 'material-ui';
import ContentFilter from 'material-ui/svg-icons/navigation/arrow-downward';
&nbsp;
const Exercises = <span class="fstat-no" title="function not covered" >({</span> exercises }) =&gt;
<span class="cstat-no" title="statement not covered" >  &lt;ToolbarGroup&gt;</span>
    &lt;ToolbarTitle style={{ color: '#EEEEEE' }} text="Exercies for today" /&gt;
    &lt;IconMenu
      value="0"
      iconButtonElement={
        &lt;IconButton&gt;
          &lt;ContentFilter color="white" /&gt;
        &lt;/IconButton&gt;
      }
    &gt;
      {!_.isEmpty(exercises)
        ? _.map(exercises, <span class="fstat-no" title="function not covered" >ex</span>ec =&gt;
<span class="cstat-no" title="statement not covered" >            &lt;MenuItem</span>
              key={exercises.indexOf(exec)}
              value={exercises.indexOf(exec)}
              primaryText={exec.toUpperCase()}
            /&gt;
          )
        : &lt;MenuItem key={0} value={0} primaryText="NO EXERCISE FOR TODAY" /&gt;}
    &lt;/IconMenu&gt;
  &lt;/ToolbarGroup&gt;;
&nbsp;
Exercises.propTypes = {
  exercises: PropTypes.arrayOf(PropTypes.string)
};
&nbsp;
export default Exercises;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
