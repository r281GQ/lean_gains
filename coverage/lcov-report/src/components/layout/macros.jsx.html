<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/layout/macros.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/layout</a> macros.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/5</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import {
  ToolbarGroup,
  ToolbarTitle,
  CircularProgress,
  RaisedButton
} from 'material-ui';
&nbsp;
import withDataCheck from './../../containers/enhancers/with_data_check';
&nbsp;
const requirements = [
  { name: 'target', path: ['userDetails', 'kcaltargets', 'waist'] },
  { name: 'dob', path: ['userDetails', 'dob'] },
  { name: 'sex', path: ['userDetails', 'sex'] },
  { name: 'waist', path: ['userDetails', 'latestMeasurements', 'waist'] },
  { name: 'rightArm', path: ['userDetails', 'latestMeasurements', 'rightArm'] },
  { name: 'chest', path: ['userDetails', 'latestMeasurements', 'chest'] },
  {
    name: 'belowBelly',
    path: ['userDetails', 'latestMeasurements', 'belowBelly']
  },
  { name: 'height', path: ['userDetails', 'latestMeasurements', 'height'] },
  {
    name: 'leftThigh',
    path: ['userDetails', 'latestMeasurements', 'leftThigh']
  },
  { name: 'hip', path: ['userDetails', 'latestMeasurements', 'hip'] },
  { name: 'leftArm', path: ['userDetails', 'latestMeasurements', 'leftArm'] },
  { name: 'belly', path: ['userDetails', 'latestMeasurements', 'belly'] },
  { name: 'weight', path: ['userDetails', 'latestMeasurements', 'weight'] },
  {
    name: 'aboveBelly',
    path: ['userDetails', 'latestMeasurements', 'aboveBelly']
  },
  {
    name: 'rightThigh',
    path: ['userDetails', 'latestMeasurements', 'rightThigh']
  },
  { name: 'neck', path: ['userDetails', 'latestMeasurements', 'neck'] }
];
const Macros = <span class="fstat-no" title="function not covered" >({</span> todaysMacros, isFetching }) =&gt;
<span class="cstat-no" title="statement not covered" >  &lt;ToolbarGroup&gt;</span>
    {isFetching
      ? &lt;CircularProgress /&gt;
      : &lt;ToolbarTitle
          style={{ color: '#EEEEEE' }}
          text={`Macros: ${todaysMacros.calorie} P: ${todaysMacros.protein} C: ${todaysMacros.carbohydrate} F: ${todaysMacros.fat}`}
        /&gt;}
  &lt;/ToolbarGroup&gt;;
&nbsp;
Macros.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  todaysMacros: PropTypes.shape({
    calorie: PropTypes.number.isRequired,
    protein: PropTypes.number.isRequired,
    carbohydrate: PropTypes.number.isRequired,
    fat: PropTypes.number.isRequired
  }).isRequired
};
&nbsp;
export default withDataCheck(
  Macros,
  requirements,
<span class="fstat-no" title="function not covered" >  ()</span> =&gt;
<span class="cstat-no" title="statement not covered" >    &lt;ToolbarGroup&gt;</span>
      &lt;RaisedButton label="aint thtat goood" /&gt;
    &lt;/ToolbarGroup&gt;,
  { name: 'target' }
);
&nbsp;
// export default Macros;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
