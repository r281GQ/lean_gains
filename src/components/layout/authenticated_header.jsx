import React from 'react';
import PropTypes from 'prop-types';
import { Toolbar } from 'material-ui';

import RouterNavigation from './router_navigation';
import Main from './main';
import Exercises from './exercises';
import User from './user';
import Macros from './macros';
import SaveIndicator from './save_indicator';



const AuthenticatedHeader = ({
  openSideBar,
  exercises,
  isLoading,
  userName,
  photo,
  todaysMacros,
  goBack,
  goForward
}) =>
  <Toolbar style={{ backgroundColor: '#757575', color: '#EEEEEE' }}>
    <Main openSideBar={openSideBar} />
    <RouterNavigation goBack={goBack} goForward={goForward} />
    <Exercises exercises={exercises} />
    <Macros todaysMacros={todaysMacros} isFetching={isLoading} />
    <SaveIndicator isLoading={isLoading} />
    <User userName={userName} photo={photo} />
  </Toolbar>;

AuthenticatedHeader.propTypes = {
  openSideBar: PropTypes.func,
  exercises: PropTypes.arrayOf(PropTypes.string),
  isLoading: PropTypes.bool,
  userName: PropTypes.string,
  photo: PropTypes.string,
  todaysMacros: PropTypes.object
};

export default AuthenticatedHeader;
