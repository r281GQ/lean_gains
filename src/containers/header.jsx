import React from 'react';
import Immutable from 'immutable';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as _ from 'lodash';
import { FlatButton, IconButton, FontIcon } from 'material-ui';
import {
  Toolbar,
  ToolbarGroup,
  ToolbarSeparator,
  ToolbarTitle,
  DropDownMenu,
  MenuItem,
  IconMenu,
  CircularProgress
} from 'material-ui';

import ActionHome from 'material-ui/svg-icons/action/home';
import Done from 'material-ui/svg-icons/action/done';
import ArrowBack from 'material-ui/svg-icons/navigation/arrow-back';
import ContentFilter from 'material-ui/svg-icons/content/filter-list';

import { goBack } from 'react-router-redux';
import currentWeightSelector from './../store/selectors/current_weight';
import isTrainingDay from './../store/selectors/exercises';
import { openSideBar } from './../store/actionCreators/app_action_creators';
import todayMacros from './../store/selectors/current_macros';

//TODO just saving indicator
const HeaderContainer = ({
  currentWeight,
  exercises,
  todaysMacros,
  openSideBar,
  isLoading,
  userName,
  goBack
}) => {
  return (
    <Toolbar>
      <ToolbarGroup firstChild={true}>
        <IconButton tooltip="Main menu" onTouchTap={() => openSideBar()}>
          <ActionHome />
        </IconButton>
      </ToolbarGroup>
      <ToolbarGroup>
        <IconButton onTouchTap={goBack} tooltip={`Go back`}>
          <ArrowBack />
        </IconButton>
      </ToolbarGroup>
      <ToolbarGroup>
        <ToolbarTitle text={`Welcome ${userName} `} />
      </ToolbarGroup>
      <ToolbarGroup>
        <ToolbarTitle text={`CurrentWeight ${currentWeight}`} />
        <ToolbarTitle text={`Macros: ${todaysMacros.calorie}`} />
        <ToolbarTitle
          text={`P: ${todaysMacros.protein} C: ${todaysMacros.carbohydrate} F: ${todaysMacros.fat}`}
        />
      </ToolbarGroup>

      <ToolbarGroup>
        {isLoading ? <CircularProgress /> : <Done />}
      </ToolbarGroup>
      <ToolbarGroup>
        <ToolbarTitle text="Exercies for today" />
        <IconMenu
          value={`0`}
          onChange={() => console.log()}
          iconButtonElement={
            <IconButton>
              <ContentFilter />
            </IconButton>
          }
        >
          {_.map(exercises, exec =>
            <MenuItem
              key={exercises.indexOf(exec)}
              value={exercises.indexOf(exec)}
              primaryText={exec}
            />
          )}
        </IconMenu>
      </ToolbarGroup>
    </Toolbar>
  );
};

const mapStateToProps = state => ({
  isLoading: state.getIn(['app', 'isLoading']),
  currentWeight: currentWeightSelector(state),
  exercises: isTrainingDay('main')(state).toJS(),
  todaysMacros: todayMacros(state).toJS(),
  userName: state.getIn(['userDetails', 'userName'])
});

const mapDispatchToProps = dispatch => ({
  goBack: () => dispatch(goBack()),
  openSideBar: () => dispatch(openSideBar())
});

export default connect(mapStateToProps, mapDispatchToProps)(HeaderContainer);
