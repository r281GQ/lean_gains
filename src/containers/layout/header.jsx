import React from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { goBack, goForward } from 'react-router-redux';
import Helmet from 'react-helmet';

import { openSideBar } from './../../store/actionCreators/app_action_creators';
import isTrainingDay from './../../store/selectors/exercises';
import todayMacros from './../../store/selectors/current_macros';
import UnauthanticatedHeader from './../../components/layout/unauth_header';
import AuthanticatedHeader from './../../components/layout/authenticated_header';
// import withDataCheck from './../enhancers/with_data_check';



const HeaderContainer = ({
  location: { pathname },
  exercises,
  todaysMacros,
  openSideBar,
  isLoading,
  userName,
  photo,
  isAuthenticated,
  goBack,
  goForward
}) =>
  <div>
    <Helmet>
      <meta charSet="utf-8" />
      <title>LeanGains</title>
      <link
        rel="icon"
        href="https://d30y9cdsu7xlg0.cloudfront.net/png/61493-200.png"
      />
      <link rel="canonical" href="http://mysite.com/example" />
    </Helmet>
    {!isAuthenticated
      ? <UnauthanticatedHeader pathname={pathname} />
      : <AuthanticatedHeader
          openSideBar={() => openSideBar()}
          todaysMacros={todaysMacros.toJS()}
          exercises={exercises.toJS()}
          isLoading={isLoading}
          userName={userName}
          photo={photo}
          goBack={goBack}
          goForward={goForward}
        />}
  </div>;

HeaderContainer.propTypes = {
  exercises: ImmutablePropTypes.list,
  todaysMacros: ImmutablePropTypes.map,
  openSideBar: PropTypes.func,
  isLoading: PropTypes.bool,
  userName: PropTypes.string,
  photo: PropTypes.string,
  isAuthenticated: PropTypes.bool
};

const mapStateToProps = state => ({
  isAuthenticated: state.getIn(['auth', 'authenticated']),
  isLoading: state.getIn(['app', 'isLoading']),
  exercises: isTrainingDay('main')(state),
  todaysMacros: todayMacros(state),
  userName: state.getIn(['userDetails', 'userName']),
  photo: state.getIn(['userDetails', 'picture'])
});

export default connect(mapStateToProps, { openSideBar, goBack, goForward })(
  withRouter(HeaderContainer)
);
