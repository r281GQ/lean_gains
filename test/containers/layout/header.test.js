import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';

import { PureHeaderContainer } from './../../../src/containers/layout/header';

describe('Header test', () => {
  it('should render', () => {
    const openSideBarSpy = sinon.spy();
    const component = shallow(
      <PureHeaderContainer
        location={{ pathname: '' }}
        openSideBar={openSideBarSpy}
      />
    );

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should call openSideBarSpy on _handleOpenSideBar', () => {
    const openSideBarSpy = sinon.spy();
    const component = shallow(
      <PureHeaderContainer
        location={{ pathname: '' }}
        openSideBar={openSideBarSpy}
      />
    );

    component.instance()._handleOpenSideBar();

    expect(openSideBarSpy.calledOnce).toBe(true);

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
