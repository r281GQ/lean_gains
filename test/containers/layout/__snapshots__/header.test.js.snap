// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should call openSideBarSpy on _handleOpenSideBar 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      LeanGains
    </title>
    <link
      href="[object Object]"
      rel="icon"
    />
  </HelmetWrapper>
  <UnauthanticatedHeader
    pathname=""
  />
</div>
`;

exports[`Header test should render 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      LeanGains
    </title>
    <link
      href="[object Object]"
      rel="icon"
    />
  </HelmetWrapper>
  <UnauthanticatedHeader
    pathname=""
  />
</div>
`;
